### PR TITLE
LPD-19196 Show scheduled document on search on management toolbar

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
changes asked on https://github.com/brianchandotcom/liferay-portal/pull/149599#issuecomment-2080579115

When searching on the management toolbar of the DM page if we search the document scheduled should be returned

1. FF LPD-10701 should be active
2. Create a document with title "test"
3. Create a scheduled document with title "test 2"
4. go to DM page
5. search by "test"

Expected:
The 2 documents should appear
[
 Compartir.pantalla.-.2024-04-22.12_42_10.mp4 ](https://github.com/liferay-content-management/liferay-portal/assets/47524751/1d577912-8529-4687-9973-f4c47893b77b)

- **LPD-19196 use injection instead of utils**
- **LPD-19196 add test to check that a recently created scheduled file is indexed and showing on the search**
- **LPD-19196 move test to another file to check more cases**
- **LPD-19196 add more tests to check behavior**
- **LPD-19196 reindex if scheduled too**
- **LPD-19196 find fileEntries by several statuses**
- **LPD-19196 buildService**
- **LPD-19196 do not create a draft assetEntry if there is an existing one**
- **LPD-19196 update the fileEntry on the case that is scheduled and the status has changed**
- **LPD-19196 update the asset if the new status is scheduled too and delete the draft if exists**
- **LPD-19196 rename**
- **LPD-19196 autoSF**

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
co:tesT:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

File: modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/search/test/DLFileEntrySearchByStatusTest.java [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+	}
+
+	@Test
+	public void testSearchFileAnyStatus() throws Exception {
+		String titlePrefix = "Document ";
+
+		_dlAppLocalService.addFileEntry(
```
Comment (jkappler) [inline modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/search/test/DLFileEntrySearchByStatusTest.java:67]: @AliciaGarciaGarcia, we could add a method to create the fileEntry, passing the display date. That would simplify a bit the code

File: modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/search/test/DLFileEntrySearchByStatusTest.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+			displayDate, null, null,
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+
+		SearchContext searchContext = SearchContextTestUtil.getSearchContext(
+			_group.getGroupId());
+
+		searchContext.setAttribute("status", WorkflowConstants.STATUS_ANY);
+		searchContext.setKeywords(titlePrefix);
+
+		Indexer<DLFileEntry> indexer = _indexerRegistry.getIndexer(
```
Comment (jkappler) [inline modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/search/test/DLFileEntrySearchByStatusTest.java:93]: @AliciaGarciaGarcia, we could move this to a method, maybe called _assertHits and pass the searchContext and the expected hits. That will also simplify a bit the code. What do you say?

**Review by jkappler — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/search/test/DLFileEntrySearchByStatusTest.java (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/search/test/DLFileEntrySearchByStatusTest.java:67]: perfect, I will create a method passing the title and the displayDate.
Thanks!

File: modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/search/test/DLFileEntrySearchByStatusTest.java (hunk 2 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/search/test/DLFileEntrySearchByStatusTest.java:93]: There is a test without the status, so I will see if I can make something.

Thanks for the suggestion.

**Comment by AliciaGarciaGarcia** [comment]
hi @jkappler 

I rebased with master and sent 2 additional commits.

Can you recheck

Thanks!

**Review by jkappler — APPROVED** [review]

**Comment by jkappler** [comment]
@AliciaGarciaGarcia

**Comment by jkappler** [comment]
ci:report:2565427200

**Comment by jkappler** [comment]
ci:forward:force

**Pull request closed without merging**